src/main/java/com/synopsys/integration/alert/web/certificates/CertificatesController.java

```diff
@@ -49,24 +54,34 @@
     public static final String API_BASE_URL = BaseController.BASE_PATH + "/certificates";
     private static final Logger logger = LoggerFactory.getLogger(CertificatesController.class);
 
+    private DescriptorKey descriptorKey;
+    private AuthorizationManager authorizationManager;
     private final ResponseFactory responseFactory;
     private final ContentConverter contentConverter;
     private final CertificateActions actions;
 
     @Autowired
-    public CertificatesController(ResponseFactory responseFactory, ContentConverter contentConverter, CertificateActions actions) {
+    public CertificatesController(CertificatesDescriptorKey descriptorKey, AuthorizationManager authorizationManager, ResponseFactory responseFactory, ContentConverter contentConverter, CertificateActions actions) {
+        this.descriptorKey = descriptorKey;
+        this.authorizationManager = authorizationManager;
         this.responseFactory = responseFactory;
         this.contentConverter = contentConverter;
         this.actions = actions;
     }
 
     @GetMapping
     public ResponseEntity<String> readCertificates() {
+        if (!hasPermission(authorizationManager::hasDeletePermission)) {
```

Comment:
Why is this checking `delete` permissions?

Reply:
fixed this